Gravatar - Update Gravatar SDK with the lastest version

hamorillo opened the pull request:
This PR updates the `gravatar-library` branch with the last version of the SDK in the artifact repository.

-----

## To Test:

Launch either the `jetpack` or `wordpress` flavor and verify that you can upload a new avatar image.

-----

hamorillo committed:
Use Gravatar SDK from A8C artifact repository

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hamorillo commented on WordPress/src/main/AndroidManifest.xml:
```diff
@@ -761,12 +761,6 @@
             android:theme="@style/WordPress.NoActionBar"
             tools:ignore="LockedOrientationActivity"/>
 
-        <!-- Lib activities-->
-        <activity
-            android:name="com.yalantis.ucrop.UCropActivity"
-            android:label="@string/edit_photo_screen_title"
-            android:theme="@style/WordPress.NoActionBar"/>
-
```
We have to remove this from the Wordpress manifest because it's also declared on the SDK manifest. To be honest, after having to do this, I think we should remove this activity declaration from the SDK and just add in the documentation that, if they want to use the upload `@Composable` they should add those lines.

hamorillo commented on build.gradle:
```diff
@@ -29,7 +29,7 @@ ext {
     wordPressUtilsVersion = '3.13.0'
     indexosMediaForMobileVersion = '43a9026f0973a2f0a74fa813132f6a16f7499c3a'
-    gravatarVersion = '1.0.0'
+    gravatarVersion = 'trunk-03c676dc112256edbd20cf2b2dc9595e7236579d'
```
We will replace this with the first released version.

hamorillo commented on config/lint/baseline.xml:
Lint is failing due to the baseline. I think it's because of all the new versions of the dependencies we are introducing along with the Gravatar SDK, but I'm not 100% sure. So, if you have a clear idea about this, please let me know.

hamorillo committed:
Fix lint issues

As Gravatar SDK is using: androidx.compose:compose-bom:2024:02:00, that makes WPAndroid use also that version, and some lint errors appear.

irfano commented:
Hey @hamorillo! What is the latest status of this PR? Does it still need to be reviewed and merged?

hamorillo commented:
> Hey @hamorillo! What is the latest status of this PR? Does it still need to be reviewed and merged?

Hi @irfano. This PR targeted a feature branch, and we no longer need it. Thanks for the ping! I totally forgot about it.
